### ISSUE TITLE
Feature Request: Resend active alerts after silencer is expired.

### ISSUE THREAD

**Issue by hatosbarbosa** [prometheus/alertmanager#2896]
Hey there! I've been using alertmanager for a couple years, and it's been a great tool, thanks for the great job you guys have done!

From what I've experienced, when a silencer is created it won't let the firing alerts to be sent to the receivers. But when the silencer is expired, the firing alerts aren't sent right away, it respects the **repeat_interval** value in the alertmanager.yml .
It makes sense to me, but:
Is there a way to force the firing alerts to be sent to the receivers right after the silencer is expired?

If there's a way to do it already, could someone explain to me how it's done?  And if there's not, can it be developed? 

I'm not a go dev, but I could try to develop a feature like this if someone is able just to guide me for a couple minutes through the project (and the current architecture allows it, of course)

Thanks!

**Comment by gotjosh** [comment]
HI @hatosbarbosa,

Thanks for reaching out. This is something that's definitely on my radar, but it's a bit more complex than that. I think that has a first step, we need to get the terminology right - right now we have concepts of muting, silencing, inhibitions, etc which kind of are related to the same thing.

Without agreement on terminology, It's impossible to know whether something like `fire waiting for notifications on `repeat_interval` if silences expire" is a breaking change or not.

I feel like this is mostly a duplicate of https://github.com/prometheus/alertmanager/issues/2496 and it's best if we keep the conversation focused around it. Feel free to drop your thoughts on that issue and please 👍  the issue as this helps us prioritise a bit better.

**Comment by juliusv** [comment]
Aside:

> it respects the repeat_interval

@hatosbarbosa you mean the `group_interval`, not the `repeat_interval`, right?

**Comment by waltherlee** [comment]
I feel the description could mean at least two things:

1. If an alert is received after a silence was created, a notification should be sent right away after the silence expired. That would be a duplicate of https://github.com/prometheus/alertmanager/issues/2496
2. Or, an alert is received **before** a silence is created and successfully sends a first notification. If the silence expires before `repeat_interval` has passed, then the alertmanager will respect the interval and wait to resend another one. @hatosbarbosa would like to have a way to resend it immediately after the silence expired.

I don't see the use case for option 2. It can be confusing for users since you expect an alert to notify again only after the interval.

I'm closing it in favor of https://github.com/prometheus/alertmanager/issues/2496 since this has been stale for a few years.

Happy to reopen it if the intention was option 2 and have more context about the use case.
